client/src/components/UserSignUpForm.js

```diff
@@ -10,6 +10,7 @@ export class UserSignUpForm extends Component {
       name: "",
       password: "",
       confirmationPassword: "",
+      role: "User",
```

Comment:
`role` is not going to change here, so it doesn't need to be part of the state. Use state only for values which are going to change after some user interactions.